aviatrix/resource_aviatrix_firewall.go

```diff
@@ -273,7 +282,18 @@ func resourceAviatrixFirewallUpdate(d *schema.ResourceData, meta interface{}) er
 
 	d.Partial(true)
 
-	log.Printf("[INFO] Creating Aviatrix firewall: %#v", firewall)
+	if d.HasChange("gw_name") {
+		_, gwNameNew := d.GetChange("gw_name")
+		gateway := &goaviatrix.Gateway{
+			GwName:         gwNameNew.(string),
+			GwOriginalName: d.Get("gw_original_name").(string),
+		}
+		err := client.IsGatewayNameUpdatable(gateway)
```

Comment:
What does this function do? Could you add a short description at its definition?

Reply:
This function is to check what value the gw_name is changed to. If it is updated to the currently used alias, it is OK. Otherwise, it is not allowed.